### PR TITLE
Improve contribution guidelines

### PR DESCRIPTION
**Pull request by tpdorsey**
## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by tpdorsey** [comment]
please review

File: docs/contributing.md [hunk 1]
```diff
@@ -30,48 +30,110 @@ If you're looking for something to get your teeth into, check out the
...
-# ... commit, push, and create your pull request
+## Documentation style
+
+Documentation is written in Markdown and built using Material for MkDocs.
+
+In general, documentation should be written in a friendly, approachable style. It should be easy to read and understand, and should be as concise as possible while still being complete.
+
+Code examples are encouraged, but should be kept short and simple. However, every code example should be complete, self-contained, and runnable. (If you're not sure how to do this, ask for help!) We prefer print output to naked asserts, but if you're testing something that doesn't have a useful print output, asserts are fine.
+
+Pydanitic's test coverage will test all code examples in the documentation, so it's important that they are correct and complete. When adding a new code example, use `pytest --update-examples` to update the output and create Python version-specific examples when appropriate.
```
Comment (madkinsz) [inline docs/contributing.md:128]:
```suggestion
Pydantic's test coverage will test all code examples in the documentation, so it's important that they are correct and complete. When adding a new code example, use `pytest --update-examples` to update the output and create Python version-specific examples when appropriate.
```

File: docs/contributing.md [hunk 2]
```diff
@@ -30,48 +30,110 @@ If you're looking for something to get your teeth into, check out the
...
+
+```bash
+# Run tests and update code examples
+'pytest --update-examples'
```
Comment (madkinsz) [inline docs/contributing.md:132]: Should this be quoted?

File: docs/contributing.md [hunk 3]
```diff
@@ -30,48 +30,110 @@ If you're looking for something to get your teeth into, check out the
...
 ```
 
-**tl;dr**: use `make format` to fix formatting, `make` to run tests and linting & `make docs`
-to build the docs.
+### Code documentation
+
+Modules, classes, functions, and module-level variables should be documented using docstrings formatted according to [PEP 257](https://www.python.org/dev/peps/pep-0257/). Type annotations should be used wherever possible.
+
+We use [Google-style docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings) for all docstrings. (See [Example Google Style Python Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for further examples.)
```
Comment (madkinsz) [inline docs/contributing.md:139]:
```suggestion
Pydantic uses [Google-style docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings) for all docstrings. (See [Example Google Style Python Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for further examples.)
```

File: docs/contributing.md (hunk 2 above)
Comment (madkinsz) [inline docs/contributing.md:132]:
```suggestion
pytest --update-examples
```

File: docs/contributing.md [hunk 4]
```diff
@@ -30,48 +30,110 @@ If you're looking for something to get your teeth into, check out the
...
 make format
 # Pydantic uses black and ruff
 # (https://github.com/ambv/black, https://github.com/charliermarsh/ruff)
 
-# 6. Run tests and linting
+# Run tests and linting
 make
-# there are a few sub-commands in Makefile like `test`, `testcov` and `lint`
+# There are a few sub-commands in Makefile like `test`, `testcov` and `lint`
 # which you might want to use, but generally just `make` should be all you need
```
Comment (dmontagu) [inline docs/contributing.md:94]:
```suggestion
# which you might want to use, but generally just `make` should be all you need.
# You can run `make help` to see more options.
```

File: docs/contributing.md [hunk 5]
```diff
@@ -30,48 +30,110 @@ If you're looking for something to get your teeth into, check out the
...
+### Build documentation
+
+If you've made any changes to the documentation (including changes to function signatures, class definitions, or docstrings that will appear in the API documentation), make sure it builds successfully.
 
-# 8. Build documentation
+```bash
+# Build documentation
 make docs
-# if you have changed the documentation make sure it builds successfully
+# If you have changed the documentation make sure it builds successfully
```
Comment (dmontagu) [inline docs/contributing.md:104]:
```suggestion
# If you have changed the documentation, make sure it builds successfully.
```

File: docs/contributing.md [hunk 6]
```diff
@@ -30,48 +30,110 @@ If you're looking for something to get your teeth into, check out the
...
 # you can also use `make docs-serve` to serve the documentation at localhost:8000
```
Comment (dmontagu) [inline docs/contributing.md:105]:
```suggestion
# You can also use `make docs-serve` to serve the documentation at localhost:8000
```

**Review by dmontagu — APPROVED** [review]

**Comment by dmontagu** [comment]
I added a couple comments, feel free to accept/reject them at your discretion and then merge when ready.

I'll ping Samuel about the cloudflare thing, I think he's going to at least get me access so I can see what's going wrong there.
